Automattic/newspack-plugin
# feat: enhancements for the Non editing contributor role

### All Submissions:






### Changes proposed in this Pull Request:



Enhances the Non-Editing Contributor role with a few new features

* If the user is only under this role, block login and remove some options from their profile
 * remove username and password, woo fields, application password, etc
 * Make email optional

* Add new UI to add this role to users under other roles

* Dont allow users with this role to delete their profiles from My Account

### How to test the changes in this Pull Request:


1. Checkout this branch
2. Add `define( 'NEWSPACK_DISABLE_CAP_GUEST_AUTHORS', true );` to wp-config
3. Visit Users > Guest Authors and see the replacement page
4. Click "Add new Non-Editing Contributor" and confirm you see the regular form to add a new user but with some modifications (no username, no password)
5. Create a user and confirm you can assign it as coauthor
6. Confirm you can not login as this user
7. Confirm the edit page for this user does not have a bunch of fields (username, password, application password, woo fields, profile colors, etc)
8. Edit a user under a subscriber-like role
9. Check the new UI and mark "Allow this user to be assigned as a co-author of a post. " - Save
10. Login as that user
11. Confirm you can't delete your account from the My Account page
12. Confirm you can assign this user as co-author


### Other information:
